### DISCUSSION TITLE
How to configure Zenoh to use over CANBus

### DISCUSSION THREAD

**Discussion by break2make** [eclipse-zenoh/roadmap#162] [Q&A]
I tried to use Zenoh over canbus with the following configuration:

zenoh-config.yaml
```
transport:
  - type: canbus
    interface: vcan0
    bitrate: 500000
```

publisher.py:

```
import zenoh
import time

# Initialize Zenoh session
session = zenoh.open(zenoh.Config.from_file("zenoh-config.yaml"))

# Publish data to a key
while True:
    session.put('/hpc/node1/data', 'High-speed data')
    print("Published: High-speed data")
    time.sleep(1)

# Close the session
session.close()
```

I am getting the following error:
zenoh.ZError: YAML error: transport: invalid type: sequence, expected struct TransportConf

What is the right way to add configuration for a new transport/network protocol? Thanks.

**Comment by break2make** [comment]
I have updated the zenoh-config.json
```
{
    "mode": "peer",
    "connect": {
        "endpoints": [
            "canbus://can0?bitrate=500000"
        ]
    }
}
```
But now getting error: `zenoh.ZError: Unsupported protocol: canbus`

**Comment by gabrik** [comment]
Hi @break2make,

Eclipse Zenoh does not have (yet) support for CANBUS.

**Comment by bestouff** [comment]
> Eclipse Zenoh does not have (yet) support for CANBUS.

Is there a beginning of an implementation somewhere ?
